plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php

```diff
@@ -134,6 +152,31 @@ protected function get_settings_for_default_section() {
 				),
 			);
 
+			$additional_fonts = array();
+			if ( wc_current_theme_is_fse_theme() && class_exists( 'WP_Font_Face_Resolver' ) ) {
+				$theme_fonts = WP_Font_Face_Resolver::get_fonts_from_theme_json();
+				if ( count( $theme_fonts ) > 0 ) {
+					foreach ( $theme_fonts as $font ) {
+						if ( ! empty( $font[0]['font-family'] ) ) {
+							$additional_fonts[ $font[0]['font-family'] ] = $font[0]['font-family'];
+						}
+					}
+					ksort( $additional_fonts );
+				}
+			}
+			$font_family = array(
+				'title'    => __( 'Font family', 'woocommerce' ),
+				'id'       => 'woocommerce_email_font_family',
+				'desc_tip' => '',
+				'default'  => 'Arial',
+				'type'     => 'select',
```

Comment:
Is it possible for each select option to be rendered in the font family it's representing?
<img width="643" alt="Screenshot 2024-11-28 at 10 30 05" src="https://github.com/user-attachments/assets/4f03977b-eedc-4c10-b4b5-de1de1466b22">

In the site editor, you see how the font looks, which makes it much easier to select the font you like.
<img width="264" alt="Screenshot 2024-11-28 at 10 29 56" src="https://github.com/user-attachments/assets/b2521c52-d07a-4c14-a30e-c6fdf0931419">
<img width="330" alt="Screenshot 2024-11-28 at 10 31 36" src="https://github.com/user-attachments/assets/774b0aac-6ce9-4445-853a-0601bfa0a2c8">

Reply:
It seems the stock Woo select components don't support this kind of customization, so I've added our own implementation of the field for that.

Reply:
@alex-mpoet could you please share how it looks for you? I see only the native OS select box, where styles are not applied.

<img width="684" alt="Screenshot 2024-11-29 at 13 07 15" src="https://github.com/user-attachments/assets/4c0d92fa-8348-4b4a-ba4c-eb65fc49540c">

Reply:
@NeosinneR You're right, that simple select-based solution had issues with rendering in Safari. I have implemented a select2-based one which is based on MailPoet's approach, it uses the `selectWoo` component already available on the Woo settings admin page. It now should render fine in Safari and other browsers.